LiteCore/Query/SQLiteDataFile+Indexes.cc

```diff
@@ -99,6 +105,20 @@ namespace litecore {
                     case IndexSpec::kFullText:
                     case IndexSpec::kVector:
                         same = schemaExistsWithSQL(indexTableName, "table", indexTableName, indexSQL);
+                        if ( spec.type == IndexSpec::kFullText ) {
+                            if ( same ) {
+                                auto whatA = spec.what(), whatB = existingSpec->what();
+                                if ( !whatA ? whatB != nullptr : !whatB ) same = false;
+                                else if ( whatA )
+                                    same = whatA->toJSONString() == whatB->toJSONString();
+                            }
+                            if ( same ) {
+                                auto whereA = spec.where(), whereB = existingSpec->where();
+                                if ( !whereA ? whereB != nullptr : !whereB ) same = false;
+                                else if ( whereA )
+                                    same = whereA->toJSONString() == whereB->toJSONString();
+                            }
+                        }
```

Comment:
Maybe a bit clearer if move this to kFullText case itself.

Reply:
You can compare Values more simply than this:
`same = v1 ? v1->isEqual(v2) : !v2;`

Reply:
This is intentional. I think the logic applies both to Vector and FTS, only at this time, I don't want to touch Vector